install/kubernetes/cilium/templates/clustermesh-apiserver/tls-cronjob/_job-spec.tpl

```diff
@@ -23,25 +21,23 @@ spec:
             {{- if .Values.debug.enabled }}
             - "--debug"
             {{- end }}
-            {{- if not (and .Values.clustermesh.apiserver.tls.ca.cert .Values.clustermesh.apiserver.tls.ca.key) }}
+            {{- if not .Values.clustermesh.apiserver.tls.ca.cert }}
             - "--clustermesh-apiserver-ca-cert-generate"
             {{- end }}
-            {{- if not (and .Values.clustermesh.apiserver.tls.server.cert .Values.clustermesh.apiserver.tls.server.key) }}
             - "--clustermesh-apiserver-server-cert-generate"
-            {{- end }}
-            {{- if not (and .Values.clustermesh.apiserver.tls.admin.cert .Values.clustermesh.apiserver.tls.admin.key) }}
             - "--clustermesh-apiserver-admin-cert-generate"
-            {{- end }}
-            {{- if not (and .Values.clustermesh.apiserver.tls.client.cert .Values.clustermesh.apiserver.tls.client.key) }}
+            {{- if .Values.externalWorkloads.enabled }}
             - "--clustermesh-apiserver-client-cert-generate"
```

Comment:
I did a quick diff, and this flag is absent in the default output in the new version. I'm not familiar enough with external workloads, so it is possible that the previous version was broken, but I wonder the change is intended or not. I can take a closer look once the other issues are addressed.

Reply:
AFAIU, `client-cert` is used by externalWorkload only. 
So I make this change to align this with tls-provided and tls-helm certgen method:
https://github.com/cilium/cilium/blob/901afa5131f27406c43a508356469141b6ed1ce0/install/kubernetes/cilium/templates/clustermesh-apiserver/client-secret.yaml#L1